src/devices/beurer_bg64.c

```diff
@@ -0,0 +1,126 @@
+ * 07 08: Impedance? reverse order ff78h = 0087h = 135 ohm? Probably needs scaling by approx 5.5.
+ * 09: Upper half nibble is checksum; invert numbers, add each nibble from 04 to 08 together, modulus 16
+ * 09: Lower half nibble is postamble, 9h (inverted)
+ * 10: Postamble, ffh inverted
+ *
+ * Beurer support, kd@beurer.de, stated these are used for the calculations on the remote display:
+ *
+ * Water:
+ *   Limitations in Anthropometric Calculations of Total Body Water, journal of the American Society of Nephrology, Copyright 2001
+ * Muscle:
+ *   Estimation of skeletal muscle mass by bioelectrical impedance analysis, J. Appl. Physiol, Copyright 2000
+ * Bone:
+ *   Body composition following hemodialysis: studies using dual energy X ray absorptiometry and bioelectrical impedance analysis.
+ *   Osteoporos Int., Copyright 1993
+ *   Abnormal body composition and reduced bone mass in growth hormone deficient hypopituitary adults.
+ *   Clin Endocrinol (Oxf), Copyright 1995
+ *
+ * However, when working through the formula for "bone", my calculations do not match what the display shows, 
+ * ~1.44% different for body fat calculations.
+  *
+ * Beurer support could/would not provide any additional clarifications;
+ *   Please note that due to our software of the scale and tolerances, deviations may occur.
+ *   Please understand that we can not provide you with any further information beyond the data mentioned above
+*/
+
+#include "rtl_433.h"
+#include "pulse_demod.h"
+#include "util.h"
+
+#define MODEL "Beurer BG64"
+
+static int beurer_bg64_callback(bitbuffer_t *bitbuffer) {
+	data_t *data;	
+	int valid = 0;
+		
+	char time_str[LOCAL_TIME_BUFLEN];
+	local_time_str(0, time_str);
+	bitrow_t *bb = bitbuffer->bb;
+	
+	//Loop through each row of data
+	for (int i = 0; i < bitbuffer->num_rows; i++)
+	{
+	        //Correct packet length?
+		if (bitbuffer->bits_per_row[i] != 88)
+			continue;
+		
+		//Valid packet? (Note; the data still needs inverting this point)
+		//The checksum calculations will check the rest of the packet
+		if ((bb[i][9] & 0x0F) != 0x9 || bb[i][10] != 0xff)
+			continue;
+
+                //Invert these numbers as we need them multiple times.
+                for (int j=0; j<=9; j++) {
+                        bb[i][j] = ~(bb[i][j]);
+                }
+
+                //Checksum, add each nibble, modulus 16 is highest nibble of column 9
+                if (( (bb[i][0] >> 4) + (bb[i][0] & 0xf) + (bb[i][1] >> 4) + (bb[i][1] & 0xf) + 
+                      (bb[i][2] >> 4) + (bb[i][2] & 0xf) + (bb[i][3] >> 4) + (bb[i][3] & 0xf) + 
+                      (bb[i][4] >> 4) + (bb[i][4] & 0xf) + (bb[i][5] >> 4) + (bb[i][5] & 0xf) + 
+                      (bb[i][6] >> 4) + (bb[i][6] & 0xf) + (bb[i][7] >> 4) + (bb[i][7] & 0xf) + 
+                      (bb[i][8] >> 4) + (bb[i][8] & 0xf) ) % 16 != (bb[i][9] >> 4) )
+		  continue;
+
+		//bitbuffer_print(bitbuffer);
+		//fprintf(stdout, "\n");

```

Comment:
Remove the 2 lines above. Except that and the changes for the unrelated lines the code is ready the be merged.

Reply:
Would rather keep the two comments as one of the nibbles, possibly battery, is not properly understood yet, and should be revisit in 3-4 months when the batteries have been utilized a little. As they are commented out, they have no adverse impact on performance.